### PR TITLE
NMS-15042: Use a "read" lock instead of an "upgrade" lock

### PR DESCRIPTION
**Pull request by j-white**
JIRA: https://issues.opennms.org/browse/NMS-15042

### CODE REVIEW

File: opennms-config/src/main/java/org/opennms/netmgt/config/SnmpPeerFactory.java [hunk 1]
```diff
@@ -109,9 +109,7 @@ public class SnmpPeerFactory implements SnmpAgentConfigFactory {
      */
     private static AtomicBoolean s_loaded = new AtomicBoolean(false);
 
-    private final ReadWriteUpdateLock m_globalLock = new ReentrantReadWriteUpdateLock();
-    private final Lock m_readLock = m_globalLock.updateLock();
```
Comment (j-white) [inline opennms-config/src/main/java/org/opennms/netmgt/config/SnmpPeerFactory.java:113]: For some reason we used an update lock here... but you can only hold 1 concurrently, so it defeats the purpose of the read lock.

**Review by RangerRick — APPROVED** [review]
seems reasonable to me; no idea why it was the other way before

**Comment by RangerRick** [comment]
is this a thing that should go into older foundations? at least 2022?

**Comment by j-white** [comment]
> is this a thing that should go into older foundations? at least 2022?

yeah, we could/should backport this as well

**Pull request merged**